### PR TITLE
feat(jira): show agile fields on issues — story points, sprints, parent, components

### PR DESCRIPTION
**Pull request by theBGuy**
This change exposes Jira Cloud agile fields in both the issues list and detail view: story points, sprint, parent/epic, components, and fix versions—each tailored per Jira site by automatically discovering the site's custom field ids. This improves issue context for teams using Scrum/Agile workflows, with zero configuration required.

### Backend: Agile Field Discovery and Extraction
- Adds `src-tauri/src/jira_field_maps.rs` for per-site discovery and caching of agile field ids (story points, sprints) using `/rest/api/3/field` and board configuration endpoints. Provides robust, tolerant storage and in-memory caching.
- Updates `src-tauri/src/forge/jira.rs`:
  - Discovers site-specific field ids and maps them into list and detail endpoints.
  - Extends both `JiraIssueInfo` and `JiraIssueDetails` to expose `storyPoints`, `sprintName`, `sprintState`, `parent`, `components`, and `fixVersions`.
  - Modularizes field extraction logic for agile fields and ensures all extractions gracefully degrade to `null`/empty if fields are missing or discovery fails.
  - Integrates a field-name map for improved field-error translation, letting error messages show field display names when available.
  - Extensively tests field extraction, naming, cache fallback, and edge cases.

- Registers the new field maps module in `src-tauri/src/lib.rs`.

### API, Documentation, and Changelog
- Documents agile fields support in `README.md`, `changelog.d/added-jira-agile-fields.md`, and help text (`src/features/help/content.ts`, `site/src/pages/index.astro`), making clear that story points, sprints, etc. appear automatically where used.
- Updates MCP server endpoint help in `src-tauri/src/mcp_server/read_jira.rs` to clarify inclusion of agile fields.

### Frontend: Exposing Agile Fields in UI
- Extends `src/lib/jira/types.ts`:
  - Adds typed fields to `JiraIssueInfo` and `JiraIssueDetails` for the new agile data.
  - Defines `JiraParentRef` and `formatStoryPoints` utility.
- `src/features/issues/IssuesPanel.tsx`: Shows story points on list rows using a pill, with accessible labeling, via the shared formatting util.
- `src/features/issues/JiraIssueView.tsx`: 
  - Shows story points, sprint name/state, parent (clickable, navigates within the Issues tab), components, and fix versions in the detail meta rows, each with precise logic to omit when empty and style consistently.
  - Groups chip sets with clear static labels for accessibility and clarity.

### CODE REVIEW

**Comment by theBGuy** [comment]
🤖 **GitDesktop AI review** · `sonnet` · automated

---

This change adds Jira agile-field support (story points, sprint, parent/epic, components, fix versions) end-to-end: per-site discovery and caching in a new Rust module, extraction helpers in `jira.rs`, and rendering in the React issue list and detail view. The overall architecture is sound — lazy discovery, two-layer caching (in-process + disk), graceful degradation to empty fields on any failure, and good unit-test coverage on the pure extraction logic. No blockers.

### Correctness

**should-fix** — `resolve_site_map` has no in-flight deduplication. If `issue_list` and `issue_view` are both called concurrently for a cold site (common at app startup, where both are inflight before either populates the cache), they each independently pass the `jira_field_maps::get` miss, pass the `DISCOVERY_FAILED` check, and each call `discover_field_map` — three network calls (`/field` + two agile calls) run twice. The second run's `put` just overwrites the first with identical data, so there's no correctness problem, but it doubles the cold-start API cost for every first-time user.

A simple fix is a per-site `OnceLock<tokio::sync::OnceCell<SiteFieldMap>>` (or an `Arc<Mutex<Option<JoinHandle<…>>>>` pattern) to coalesce concurrent discovers for the same site into a single in-flight request.

### Caching design

**nit** — `resolved_at` is persisted in `jira-field-maps.json` but never read back by any code path. The field looks like infrastructure for a future TTL check (and would be a natural place to hang a "re-discover after N days" rule), but right now the comment in the module doc ("a site legitimately without agile fields caches an all-`None` entry so we don't re-probe") silently implies the only cache invalidation is a process restart + deleted file. If a Jira admin ever moves the estimation field to a different custom field, the persisted ID will be permanently wrong. At minimum, the doc should acknowledge the absence of TTL and explain why it's acceptable (field IDs are stable in Jira). Adding a 7-day staleness check on `resolved_at` would be the proper fix but isn't urgently needed.

### Frontend

**nit** — `JiraIssueView.tsx` line:
```tsx
onClick={() =>
  issue.parent &&
  selectIssue({ kind: "jira", id: issue.parent.key })
}
```
The `issue.parent &&` guard is dead — this `<button>` only renders inside `{issue.parent && (<button …>)}`, so `issue.parent` is always truthy here. The plain `onClick={() => selectIssue({ kind: "jira", id: issue.parent!.key })}` (or rely on the outer guard and access `issue.parent.key` directly without the `&&`) reads more clearly.

---

_Posted by [GitDesktop](https://gitdesktop.app) — AI output, verify before acting on it._

**Comment by theBGuy** [comment]
🤖 **GitDesktop AI security audit** · `sonnet` · automated

---

No security issues found in these changes.

The areas I investigated and cleared:

**Field IDs embedded in URL query strings** (`detail_custom_fields_suffix` → `issue_view` URL): `story_points_field_id` and `sprint_field_id` values discovered from Jira's `/field` API are appended directly to the request URL without encoding. A malicious Jira server could return an id like `customfield_10016&extra=param`, injecting additional query parameters. However, the request still targets the same Jira host (the host is fixed by stored credentials via `resolve`/`resolve_agile`), so the attacker can't redirect to a different host and can only influence a request going back to their own server — no cross-privilege-boundary impact, does not clear the SSRF bar.

**Jira field names in error translation**: Field display names from `/field` responses populate the in-process name map and surface in error strings. React renders these as escaped text (no `dangerouslySetInnerHTML`), so no XSS path.

**`project_key` in board URL**: URL-encoded via `encode_query_value` before being embedded in `board?projectKeyOrId=…`, and the project key is grammar-validated by `build_list_jql` before `resolve_site_map` is reached.

**`board_id` in agile URL**: Typed as `i64`, not string-interpolated without bounds.

**Persisted `SiteFieldMap`**: Read from a local app-data file via tolerant serde deserialization. Any local file modification requires local write access (out of scope), and the values only reach URL query strings and JSON key lookups — both already cleared above.

**React rendering**: All Jira-origin data (`sprintName`, `sprintState`, `components`, `fixVersions`, `parent.key`, `parent.summary`) is rendered in JSX as plain text or `title`/`aria-label` attributes — all escaped by React's default behavior.

Copilot's finding about the name map staying cold after an app restart (disk-hit path skips `set_name_map`) is a real correctness gap — custom field keys won't translate to friendly names in error messages after process restart — but it is not a security issue.

---

_Posted by [GitDesktop](https://gitdesktop.app) — AI output, verify before acting on it._

File: src-tauri/src/forge/jira.rs [hunk 1]
```diff
@@ -1025,6 +1217,314 @@ const LIST_FIELDS: &[&str] = &[
...
+/// request proceeds with the skeleton fields and NEVER errors. `project_key` drives the
+/// board-config override (empty skips it).
+async fn resolve_site_map(
+    creds: &JiraCredentials,
+    site: &str,
+    project_key: &str,
+) -> crate::jira_field_maps::SiteFieldMap {
+    if let Some(entry) = crate::jira_field_maps::get(site) {
+        return entry;
+    }
```
[the remark this answers is not in the document]
Comment (theBGuy) [reply src-tauri/src/forge/jira.rs:1543]: **Applied** — went with your second option (persist a minimal id→name map). The per-site entry in `jira-field-maps.json` now carries `fieldNames` (filtered to `customfield_*` entries, captured from the same `/field` response at discovery), and `jira_field_maps::get()` hydrates the in-process name map from disk on the persisted-hit path — so error translation is warm after app restarts *and* in the headless MCP process, with zero extra network. A legacy entry without `fieldNames` decodes as absent and self-heals via one idempotent rediscovery. While in there, field ids are now also grammar-validated (`^customfield_[0-9]+$`) at discovery and on persisted load. Covered by `decode_entry_hydration_warms_translation` / `decode_entry_without_field_names_is_absent`; suite at 599 passing.

---

_Posted by [GitDesktop](https://gitdesktop.app) — automated agent comment, verify before acting on it._

File: src/features/issues/JiraIssueView.tsx [hunk 2]
```diff
@@ -519,6 +523,37 @@ export function JiraIssueView({
           {issue.resolutionName && (
             <span>Resolution: {issue.resolutionName}</span>
           )}
+          {issue.storyPoints != null && (
+            <span>Points: {formatStoryPoints(issue.storyPoints)}</span>
+          )}
```
[the remark this answers is not in the document]
Comment (theBGuy) [reply src/features/issues/JiraIssueView.tsx:528]: **Applied** — the detail meta row now reads `Story points: 3`, matching the list-row tooltip, docs, and site copy.

---

_Posted by [GitDesktop](https://gitdesktop.app) — automated agent comment, verify before acting on it._

**Comment by theBGuy** [comment]
**Round-1 review triage** — all three reviews (Copilot, AI review, AI security audit) verified against the code; dispositions:

- **`resolve_site_map` in-flight dedup (AI review, should-fix): applied.** Discovery is now coalesced per site — a per-site async lock with a double-checked cache re-read, so racers await the winner's result instead of each probing `/field`. Different sites don't serialize against each other, and no std `MutexGuard` spans an `.await` (`await_holding_lock` clean).
- **`resolvedAt` unread / no TTL (AI review, nit): applied as documentation.** The module doc now states the deliberate design: no TTL or invalidation — Jira field ids are stable, a site that later moves its estimation field re-resolves only when the cache file is deleted, and `resolvedAt` is informational today (the natural hook for a future staleness rule).
- **Dead `issue.parent &&` guard (AI review, nit): declined, with evidence.** The guard is not dead — TS narrowing from the outer `{issue.parent && …}` does not extend into a callback, and removing it fails the build with `TS18047: 'issue.parent' is possibly 'null'`. It satisfies the compiler, not runtime logic; an in-code comment at the site now says exactly that.
- **Cold name map + `Points:` label (Copilot): applied** — details in the inline thread replies (persisted `fieldNames` + disk hydration; `Story points:` label).
- **Unencoded field ids in URLs (security audit, cleared): hardened anyway.** The audit correctly cleared it (same-host, no privilege crossing), but ids are now grammar-validated `^customfield_[0-9]+$` at discovery *and* on persisted load — a hostile `customfield_10016&extra=1` is rejected at both layers (unit-tested).

Verification on the combined tree: `cargo test` 599 passed / 0 failed, `cargo clippy -- -D warnings` clean, `pnpm build` green.

---

_Posted by [GitDesktop](https://gitdesktop.app) — automated agent comment, verify before acting on it._

**Comment by theBGuy** [comment]
🤖 **GitDesktop AI review** · `sonnet` · automated

---

Round 2 addresses all three previous findings cleanly: the per-site `tokio::sync::Mutex` coalescing lock (with correct double-checked re-read after acquiring) eliminates redundant cold-start discovery; the no-TTL policy is now explicitly documented in the module docs; and the `issue.parent &&` guard in the callback is explained with a comment. The Copilot cold-name-map finding is also addressed — `get` now calls `set_name_map` from the persisted `fieldNames` on every disk hit, and there's a test confirming the hydration path. Architecture is sound; no blockers.

### Resolved since last review

- **In-flight dedup** (`resolve_site_map`): per-site `Arc<tokio::sync::Mutex<()>>` with double-checked cache re-read confirmed at lines 1556–1563. The std registry mutex is released before any `.await`, so no `await_holding_lock` violation.
- **No TTL/`resolvedAt` unread**: explicit "No TTL / invalidation — by design" section added to module docs with a rationale and a clear statement of `resolvedAt` as an informational/future hook.
- **Dead `issue.parent &&`**: comment at the call site correctly explains the TS18047 narrowing constraint.
- **Cold name map on restart** (Copilot): `get` hydrates `set_name_map` from the persisted `fieldNames`; legacy entries (`field_names: None`) decode to `None` in `decode_persisted_entry`, triggering one idempotent rediscovery; test `decode_entry_without_field_names_is_absent` covers this.

### Nit

**nit** — `is_customfield_key` (private, `jira.rs` line 222) and `is_valid_field_id` (pub, `jira_field_maps.rs` line 75) are byte-for-byte identical. `is_valid_field_id` is already called from `jira.rs` three times elsewhere (`resolve_field_ids`, `field_name_map`, `board_config_points_override`), and the unit test `is_customfield_key_matches_only_numeric_customfields` duplicates the coverage in `jira_field_maps`'s test suite. Replace `is_customfield_key` with a direct call to `crate::jira_field_maps::is_valid_field_id` in `translate_field_key`, and drop the redundant test.

---

_Posted by [GitDesktop](https://gitdesktop.app) — AI output, verify before acting on it._

**Comment by theBGuy** [comment]
**Round-2 triage** — the one nit is confirmed and applied: `is_customfield_key` was a pre-round-1 relic made redundant when `is_valid_field_id` landed with the hardening pass. `translate_field_key` now calls `crate::jira_field_maps::is_valid_field_id` directly; the private duplicate and its duplicate test are deleted (net −19 lines). Coverage stays with `is_valid_field_id_grammar` in `jira_field_maps`. Verification: `cargo test` 598 passed / 0 failed, `cargo clippy -- -D warnings` clean.

---

_Posted by [GitDesktop](https://gitdesktop.app) — automated agent comment, verify before acting on it._

**Comment by theBGuy** [comment]
🤖 **GitDesktop AI review** · `sonnet` · automated

---

The only substantive change in this round is the resolution of the outstanding nit: `is_customfield_key` is removed from `jira.rs`, its call site in `translate_field_key` now calls `crate::jira_field_maps::is_valid_field_id` directly, the doc comment is updated to name the canonical predicate, and the duplicate test is dropped. The substitution is correct — `is_valid_field_id` and the deleted function were byte-for-byte identical, `jira_field_maps`'s test suite covers the same cases, and no other call site referred to `is_customfield_key`.

### Resolved since last review

- **Duplicate `is_customfield_key` / redundant test**: removed; `translate_field_key` now delegates to `crate::jira_field_maps::is_valid_field_id`. Confirmed in the current diff.

No new findings. The PR is clean.

---

_Posted by [GitDesktop](https://gitdesktop.app) — AI output, verify before acting on it._

**Comment by theBGuy** [comment]
🤖 **GitDesktop AI review** · `sonnet` · automated

---

The only change since the last review is the cleanup of the `is_customfield_key` duplicate: the private function has been deleted, its call site in `translate_field_key` replaced with `crate::jira_field_maps::is_valid_field_id`, and the now-redundant test removed. This is exactly the fix the previous nit called for.

### Resolved since last review

- **`is_customfield_key` / `is_valid_field_id` duplication** (previous nit): confirmed resolved. `is_customfield_key` and its test `is_customfield_key_matches_only_numeric_customfields` are deleted; `translate_field_key` now delegates to the canonical `crate::jira_field_maps::is_valid_field_id`, which is `pub` and already covered by the `jira_field_maps` test suite.

No new issues introduced. The rest of the diff is unchanged from round 2, which was already cleared. This is ready to merge.

---

_Posted by [GitDesktop](https://gitdesktop.app) — AI output, verify before acting on it._
